spec/acceptance/docker_full_spec.rb

```diff
@@ -0,0 +1,643 @@
+require 'spec_helper_acceptance'
+
+describe 'Temp OS hacks' do
+  if fact('operatingsystem') =~ /CentOS/
+    if fact('operatingsystemmajrelease') == '7'
+        pp=<<-EOS
+              package {'device-mapper':
+                ensure => latest,
+              }
+        EOS
+      apply_manifest(pp, :catch_failures => true)
+    end
+  elsif default['platform'] =~ /el-7/
+      pp=<<-EOS
+            package {'device-mapper':
+              ensure => latest,
+            }
+
+            package {'docker':
+              install_options => ['--enablerepo=rhel7-extras'],
+              ensure => latest,
+              require => Package['device-mapper']
+            }
+      EOS
+      apply_manifest(pp, :catch_failures => true)
+  end
+end
+
+describe 'docker full acceptance tests section 1 - testing class {\'docker\':}' do
+  context 'test 1.1 - Install and configure Docker and Docker Daemon' do
+    it 'applies the manifest' do
+      pp=<<-EOS
+        class { 'docker': }
+      EOS
+
+      apply_manifest(pp, :catch_failures => true)
+      unless fact('selinux') == 'true'
+        apply_manifest(pp, :catch_changes => true)
+      end
+    end
+
+    it ': ps -aux should be running docker process' do
+      shell('ps -aux | grep docker') do |r|
+        expect(r.stdout).to match(/\/usr\/bin\/docker/)
+      end
+    end
+    it ': docker ps should output table headers' do
```

Comment:
This is quite brittle, given that the output of ps can and does change. I'd maybe just change to to check for a zero exit code.